esp/logging/logginglib/logthread.cpp

```diff
@@ -646,39 +646,62 @@ void CLogRequestReader::removeUnknownAckedLogFiles(StringArray& ackedLogFiles)
     ESPLOG(LogMax, "#### The removeUnknownAckedLogFiles() done.");
 }
 
+void CLogRequestReader::addNewAckedFileList(const char* list, StringArray& fileNames)
+{
+    OwnedIFile newList = createIFile(list);
+    if (!newList)
+        throw makeStringExceptionV(EspLoggingErrors::UpdateLogFailed, "Failed to access %s", list);
+    if (newList->exists())
+    {
+        newList->remove();
+        newList.setown(createIFile(list));
```

Comment:
If createIFile could have failed above, should failures be handled here?

Reply:
Yes, should throw an exception here.